frontend/src/pages/EditDepartment/EditDepartment.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+
+import { useParams } from "react-router-dom";
+import { Container, Row, Col, Breadcrumb, Form, Button, Card, Alert } from "react-bootstrap";
+import { LinkContainer } from "react-router-bootstrap";
+
+// Services
+import { getHospital } from "../../Services/HospitalService";
+import { getDepartment } from "../../Services/DepartmentService";
+import { getWardTypes, addWard, deleteWard } from "../../Services/WardService";
+
+// Types
+import { Ward, WardType } from "../../datatypes/ListView.types";
+
+// Components
+import CardData from "../../components/CardData/CardData";
+
+const EditDepartment: React.FunctionComponent = () => {
+  const [hospitalData, setHospitalData] = React.useState<any | null>(null);
+  const [departmentData, setDepartmentData] = React.useState<any | null>(null);
+  const [wardTypesList, setWardTypesList] = React.useState<any | null>(null);
+  const [wardName, setWardName] = React.useState("");
+  const [wardType, setWardType] = React.useState<WardType>();
+
+  const { hospitalId, departmentId } = useParams();
+
+  React.useEffect(() => {
+    if (hospitalId && departmentId) {
+      (async () => {
+        const hospitalResponse = await getHospital(hospitalId);
+        const departmentResponse = await getDepartment(departmentId);
+        const wardTypesResponse = await getWardTypes();
+
+        setHospitalData(hospitalResponse);
+        setDepartmentData(departmentResponse);
+        setWardTypesList(wardTypesResponse);
+        setWardType(wardTypesResponse[0].name as WardType);
+      })();
+    }
+  }, []);
```

Comment:
Should we add hospitalId and departmentId as dependencies to this useEffect function?